docs/configuration.md

```diff
@@ -0,0 +1,228 @@
+```nix
+networking.hostName = "myhost";
+time.timeZone = "UTC";
+```
+
+## Debug your configuration
+
+To print the values of specific options of your config, add the following to your `configuration.nix`:
+```nix
+system.extraDependencies = let
+  debugVal = config.networking.firewall.allowedTCPPorts;
+  # More example options:
+  # debugVal = config.environment.systemPackages;
+  # debugVal = config.services.bitcoind.dataDir;
+in lib.traceSeqN 3 debugVal [];
+```
+and run command `eval-config` from the deployment shell.
+
+# Allowing external connections to services
+
+## Allow peer connections to bitcoind
+
+```nix
+services.bitcoind = {
+  # Accept incoming peer connections
+  listen = true;
+
+  # Listen to connections on all interfaces
+  address = "0.0.0.0";
+
+  # Set this to also add IPv6 connectivity.
+  extraConfig = ''
+    bind=::
+  '';
+
+  # If you're using the `secure-node.nix` template, set this to allow non-Tor connections
+  # to bitcoind
+  tor.enforce = false;
+  # Also set this if bitcoind should not use Tor for outgoing peer connections
+  tor.proxy = false;
+};
+
+# Open the p2p port in the firewall
+networking.firewall.allowedTCPPorts = [ config.services.nix-bitcoin.port ];
+```
+
+## Allow bitcoind RPC connections from LAN
+
+```nix
+services.bitcoind = {
+  # Listen to connections on all interfaces
+  address = "0.0.0.0";
+
+  # Allow RPC connections from external addresses
+  rpc.allowip = [
+    "10.10.0.0/24" # Allow a subnet
+    "10.50.0.3" # Allow a specific address
+    "0.0.0.0" # Allow all addresses
+  ];
+
+  # Set this if you're using the `secure-node.nix` template
+  tor.enforce = false;
+};
+
+# Open the RPC port in the firewall
+networking.firewall.allowedTCPPorts = [ config.services.nix-bitcoin.rpc.port ];
+```
+
+## Allow connections to electrs
+
+```nix
+services.electrs = {
+  # Listen to connections on all interfaces
+  address = "0.0.0.0";
+
+  # Set this if you're using the `secure-node.nix` template
+  tor.enforce = false;
+};
+
+# Open the electrs port in the firewall
+networking.firewall.allowedTCPPorts = [ config.services.electrs.port ];
+```
+
+You can use the same approach to allow connections to other services.
+
+# Migrate existing services to nix-bitcoin
+
+## Example: bitcoind
+
+```shell
+# 1. Stop bitcoind on your nodes
+ssh root@nix-bitcoin-node 'systemctl stop bitcoind'
+# Also stop bitcoind on the node that you'll be copying data from
+
+# 2. Copy the data to the nix-bitcoin node
+# Important: Add a trailing slash to the source path
+rsync /path/to/existing/bitcoind-datadir/ root@nix-bitcoin-node:/var/lib/bitcoind
+
+# 3. Fix data dir permissions on the nix-bitcoin node
+ssh root@nix-bitcoin-node 'chown -R bitcoin: /var/lib/bitcoind'
+
+# 4. Start bitcoind
+ssh root@nix-bitcoin-node 'systemctl start bitcoind'
+```
+
+You can use the same workflow for other services.\
+The default data dir path is `/var/lib/<service>` for all services.\
+Some services require extra steps:
```

Comment:
JoinMarket also requires copying the wallet password

Reply:
Fixed.